src/components/radio-button/radio-button-group/radio-button-group.test.tsx

```diff
@@ -0,0 +1,292 @@
+        <RadioButton key="radio2" value="radio2" label="Radio Button 2" />,
+      ]}
+    </RadioButtonGroup>
+  );
+
+  expect(screen.getAllByRole("radio")).toHaveLength(2);
+  expect(screen.getByText("foo")).toBeVisible();
+});
+
+test("renders fieldset with provided `legend`", () => {
+  render(
+    <RadioButtonGroup
+      name="group"
+      legend="Radio Group Legend"
+      onChange={() => {}}
+    >
+      <RadioButton value="radio1" label="Radio Button 1" />
+    </RadioButtonGroup>
+  );
+
+  const fieldset = screen.getByRole("group", { name: "Radio Group Legend" });
+
+  expect(fieldset).toBeVisible();
+});
+
+test("checks the correct `RadioButton` child when a value is passed", () => {
+  render(
+    <RadioButtonGroup name="group" value="radio2" onChange={() => {}}>
+      <RadioButton value="radio1" label="Radio Button 1" />
+      <RadioButton value="radio2" label="Radio Button 2" />
+    </RadioButtonGroup>
+  );
+
+  const radio1 = screen.getByRole("radio", { name: "Radio Button 1" });
+  const radio2 = screen.getByRole("radio", { name: "Radio Button 2" });
+
+  expect(radio1).not.toBeChecked();
+  expect(radio2).toBeChecked();
+});
+
+test("calls `onChange` when a `RadioButton` child is checked", async () => {
+  const user = userEvent.setup();
+  const onChange = jest.fn();
+  render(
+    <RadioButtonGroup name="group" onChange={onChange}>
+      <RadioButton value="radio1" label="Radio Button 1" />
+    </RadioButtonGroup>
+  );
+
+  const radioButton = screen.getByRole("radio");
+
+  await user.click(radioButton);
+
+  expect(onChange).toHaveBeenCalled();
+});
+
+test("calls `onBlur` when a `RadioButton` child is blurred", async () => {
+  const user = userEvent.setup();
+  const onBlur = jest.fn();
+  render(
+    <RadioButtonGroup name="group" onChange={() => {}} onBlur={onBlur}>
+      <RadioButton value="radio1" label="Radio Button 1" />
+    </RadioButtonGroup>
+  );
+
+  await user.tab();
+  await user.tab();
+
+  expect(onBlur).toHaveBeenCalled();
+});
+
+test("renders required `RadioButton` children when `required` prop is true", () => {
+  render(
+    <RadioButtonGroup name="group" required onChange={() => {}}>
+      <RadioButton value="radio1" label="Radio Button 1" />
+    </RadioButtonGroup>
+  );
+
+  const radioButton = screen.getByRole("radio", { name: "Radio Button 1" });
+
+  expect(radioButton).toBeRequired();
+});
+
+test("renders with inline legend when screen is larger than `adaptiveLegendBreakpoint`", () => {
+  mockMatchMedia(true);
+  render(
+    <RadioButtonGroup
+      name="group"
+      legend="Radio Group Legend"
+      adaptiveLegendBreakpoint={450}
+      onChange={() => {}}
+    >
+      <RadioButton value="radio1" label="Radio Button 1" />
+    </RadioButtonGroup>
+  );
+
+  const legend = screen.getByTestId("legend");
+
+  expect(legend).toHaveStyle({ float: "left" });
+});
+
+test("removes provided margin-left when screen is smaller than `adaptiveSpacingBreakpoint`", () => {
```

Comment:
**suggestion**: as above, I'd like to cover the other case (it took me a while to realise there were two different breakpoint props being tested in these 2 tests - I thought at first they were testing the 2 different cases of a single breakpoint prop)